File: Algorithms/Easy/ContainsDuplicate.go
package easy

// Time complexity - O(n)
// Space complexity - O(n)
func containsDuplicate(nums []int) bool {
	dict := make(map[int]struct{}, len(nums))
	for _, num := range nums {
		if _, ok := dict[num]; ok {
			return true
		}

		dict[num] = struct{}{}
	}

	return false
}

File: Algorithms/Easy/217.Contains-Duplicate/ContainsDuplicate.py
# Time complexity - O(n)
# Space complexity - O(n)
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        exists = set()
        for num in nums:
            if num not in exists:
                exists.add(num)
            else:
                return True

        return False